### PR TITLE
Allow state migration on pre-gingerbread block

### PR DESCRIPTION
**Pull request by piersy**
Added flags to define gas limit and base fee if they are missing.
This allows migration to work on a pre-gingerbread block.

### CODE REVIEW

File: op-chain-ops/cmd/celo-migrate/main.go [hunk 1]
```diff
@@ -68,6 +68,14 @@ var (
 		Name:  "dry-run",
 		Usage: "Dry run the upgrade by not committing the database",
 	}
+	gasLimitFlag = &cli.Uint64Flag{
```
Comment (palango) [inline op-chain-ops/cmd/celo-migrate/main.go:71]: Do we need to make this configurable? My understanding was that this is just for testing

File: op-chain-ops/cmd/celo-migrate/main.go [hunk 2]
```diff
@@ -207,7 +217,7 @@ func main() {
 	log.Info("Finished migration successfully!")
 }
 
-func ApplyMigrationChangesToDB(genesis *core.Genesis, dbPath string, commit bool) (*types.Header, error) {
+func ApplyMigrationChangesToDB(ctx *cli.Context, genesis *core.Genesis, dbPath string, commit bool) (*types.Header, error) {
```
Comment (palango) [inline op-chain-ops/cmd/celo-migrate/main.go:220]: Ideally we'd avoid passing in the context here, could we set a default?

File: op-chain-ops/cmd/celo-migrate/main.go (hunk 1 above)
Comment (piersy) [reply op-chain-ops/cmd/celo-migrate/main.go:71]: This allows you to migrate at pre gingerbread blocks, the gas limit changed a few times pre-gingerbread so I figured that people may want to set a variety of gas limits, depending on what block the transition occurred at.

In the real migration cases we will be migrating post gingerbread so people won't need to set the flag.

File: op-chain-ops/cmd/celo-migrate/main.go (hunk 2 above)
Comment (piersy) [reply op-chain-ops/cmd/celo-migrate/main.go:220]: What's the issue with passing the context here?

File: op-chain-ops/cmd/celo-migrate/main.go (hunk 1 above)
Comment (palango) [reply op-chain-ops/cmd/celo-migrate/main.go:71]: I just wonder if setting this is important for testing? Does it matter if you can set the gas limit to 20M or if it is hardcoded to 10M? As you write in the real migration this won't be used.

File: op-chain-ops/cmd/celo-migrate/main.go (hunk 2 above)
Comment (palango) [reply op-chain-ops/cmd/celo-migrate/main.go:220]: It's not wrong but has a weird taste. If we do this we wouldn't need to pass the dbPath in as a parameter either, but there we do all handling of options before calling the migration code.

Comment (piersy) [reply op-chain-ops/cmd/celo-migrate/main.go:220]: Ok got you

File: op-chain-ops/cmd/celo-migrate/main.go (hunk 1 above)
Comment (piersy) [reply op-chain-ops/cmd/celo-migrate/main.go:71]: It's hard to say, might depend on the kind of testing you need to do. But I can just revert to a hardcoded value it works for my current use case.

What about the base fee flag, I guess you feel the same?

**Review by palango — APPROVED** [review]
Ok, if you think the flags might be useful during testing let's add them. 

Can we remove them again once the sync testing is done?

**Comment by piersy** [comment]
Hmm totally forgot about this one, closing as it has been superseded by https://github.com/celo-org/optimism/pull/179

**Pull request closed without merging**